errai-bus/src/main/java/org/jboss/errai/bus/client/framework/ClientMessageBusImpl.java

```diff
@@ -945,28 +945,53 @@ private void sendAllDeferred() {
     if (!deferredMessages.isEmpty())
       logger.info("transmitting deferred messages now ...");
 
+    List<Message> localMessages = new ArrayList<>();
```

Comment:
So be low it looks like we are already removing delivered message from `deferredMessages` as we process them. Could we possibly clean this up a bit by getting rid of this field and iterating through the remaining `deferredMessages` instead?

Reply:
Good idea, will just have to replace the `while` loop with a `for` loop.